File: tests/parsers/samples/code/py/dynamic.py
"""Dynamic entry point that loads the simple workflow at runtime."""

import importlib
from typing import Callable


def load_workflow(name: str) -> Callable[[], dict[str, float]]:
    module = importlib.import_module("simple")
    return getattr(module, name)


REGISTRY = {
    "simple_flow": load_workflow("simple_flow"),
    "summarize": load_workflow("summarize"),
}


def run_dynamic(name: str = "summarize") -> str:
    action = REGISTRY[name]
    result = action()
    if isinstance(result, dict):
        return f"dynamic:{result['baseline']:.1f}:{result['rms']:.1f}"
    return str(result)

File: tests/parsers/samples/code/zig/dynamic.zig
const std = @import("std");
const simple = @import("simple.zig");

pub fn runDynamic(allocator: std.mem.Allocator, name: []const u8) ![]u8 {
    if (std.mem.eql(u8, name, "summarize")) {
        const metrics = simple.summarize();
        return std.fmt.allocPrint(allocator, "dynamic:{d:.2}:{d:.2}", .{ metrics.baseline, metrics.rms });
    }
    return std.fmt.allocPrint(allocator, "unknown", .{});
}
